common/Models/ExtensionAdaptiveCard.cs

```diff
@@ -36,7 +49,16 @@ public ProviderOperationResult Update(string templateJson, string dataJson, stri
         // an empty string.
         var adaptiveCardString = template.Expand(Newtonsoft.Json.JsonConvert.DeserializeObject<Newtonsoft.Json.Linq.JObject>(dataJson ?? DataJson));
 
-        var parseResult = AdaptiveCard.FromJsonString(adaptiveCardString);
+        AdaptiveCardParseResult parseResult;
+
+        if (_actionParserRegistration != null && _elementParserRegistration != null)

```

Comment:
Maybe simpler: Have one constructor where the two *ParserRegistrations are optional, and do like 
```
_elementParserRegistration = elementParserRegistration != null ? elementParserRegistration : new AdaptiveElementParserRegistration();
``
Then you don't need the if/else here or in ExtensionAdaptiveCardPanel.cs, I think. Also if in the future there was a case where we had only a custom action parser or only element parser, it would handle that.

Reply:
I'll try that out